UIforETW/Settings.cpp

```diff
@@ -143,11 +143,15 @@ void CSettings::OnBnClickedCopystartupprofile()
 	std::wstring source = exeDir_ + fileName;
 
 	wchar_t documents[MAX_PATH];
-	if (!SHGetSpecialFolderPath(*this, documents, CSIDL_MYDOCUMENTS, TRUE))
+
+	const BOOL getMyDocsResult = SHGetSpecialFolderPathW(m_hWnd, documents, CSIDL_MYDOCUMENTS, TRUE);

```

Comment:
> But mostly I'm annoyed that m_hWnd is a protected member variable. So much for encapsulation.

On the other hand, what would "encapsulation" look like here? `HWND getHWND() const;`? What would that encapsulate? 

Personally, I think that simple getters & setters aren't really encapsulation, and where they're really needed, it's a sign that objects are too highly coupled, and far too big.

Maybe it's a bit unrealistic - I really don't know. Maybe it's just my inner C programmer speaking.

> At least using the conversion operator doesn't advertise the fact that m_hWnd is completely accessible

Well, yeah it doesn't advertise that fact, but it sure is surprising that a `CSettings` object is implicitly convertible to something else, partly because it doesn't advertise anything.

> CString conversion operator to const char*.

`CString` is pretty horrible, in all sorts of ways. No move constructors, passing by value does expensive reference counting, weird buffer locking/unlocking mechanisms, generally bad memory management, etc..... I've really developed a deep dislike of it.

----what follows is a bit of a rant, sorry!----
The best example of a good use of encapsulation that comes to mind is something from my work on WinDirStat.

"Child" objects were heap allocated and stored in a vector of pointers (i.e. `std::vector<CItem*>`), and those items were displayed in a list view, sorted by size, date, or name (etc...).

When I changed everything to allocate an array of objects and use placement new to construct them (which saved a lot of space, and was faster),  I couldn't sort them anymore, because then I'd actually have to move the location of the objects in memory, which would require fixing a ton of pointers to point to the new location.

Instead of crazy complex pointer juggling, I wrote a "getter" that'd actually build a vector of pointers to the array of objects, and return that- i.e.:

```
std::vector<CItem*> items;
items.reserve( childCount );
for( size_t i = 0; i < childCount; ++i ) {
    items.emplace_back(children + i);
    }
return items;
```

...it actually did some work, rather than just return the value of the internal representation.